packages/instantsearch.js/src/lib/utils/hydrateSearchClient.ts

```diff
@@ -80,40 +90,36 @@ export function hydrateSearchClient(
   // a single-index result. You can find more information about the
   // computation of the key inside the client (see link below).
   // https://github.com/algolia/algoliasearch-client-javascript/blob/c27e89ff92b2a854ae6f40dc524bffe0f0cbc169/src/AlgoliaSearchCore.js#L232-L240
-  if (!client.transporter) {
+  if (!('transporter' in client)) {
     const cacheKey = `/1/indexes/*/queries_body_${JSON.stringify({
-      requests: Object.keys(results).reduce(
-        (acc, key) =>
-          acc.concat(
-            results[key].rawResults.map((request) => ({
-              indexName: request.index,
-              params: request.params,
-            }))
-          ),
-        []
-      ),
+      requests: cachedRequest,
     })}`;
 
-    client.cache = {
-      ...client.cache,
+    (client as ClientWithCache).cache = {
+      ...(client as ClientWithCache).cache,
       [cacheKey]: JSON.stringify({
-        results: Object.keys(results).reduce(
-          (acc, key) => acc.concat(results[key].rawResults),
-          []
-        ),
+        results: Object.keys(results).map((key) => results[key].results),
       }),
     };
   }
 }
 
+function deserializeQueryParameters(parameters: string) {
+  return parameters.split('&').reduce<Record<string, any>>((acc, parameter) => {
+    const [key, value] = parameter.split('=');
+    acc[key] = value ? decodeURIComponent(value) : '';
```

Comment:
don't think it's worth deviating from the design, but instead of decode/encode uri component, we could simplify it and leave params uri encoded

Reply:
Actually I can't as it's also used in the mocked `client.search` which has decoded params, or I'd have to put a boolean or something. Don't know if it's worth it.